src/components/ShippingLabels/ShippingLabel.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import supabase, { Schema } from "@/supabase";
+import ShippingLabelsButton from "@/components/ShippingLabels/ShippingLabelsButton";
+import { ParcelClients } from "@/components/ShippingLabels/ShippingLabelsPdf";
+
+const formatDatetime = (datetimeString: string | null, isDatetime: boolean): string => {
+    if (datetimeString === null) {
+        return "No recorded date";
+    }
+
+    const dateOptions: Intl.DateTimeFormatOptions = {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "2-digit",
+        minute: "2-digit",
+    };
+
+    const formattedDate = isDatetime
+        ? new Date(datetimeString).toLocaleString([], dateOptions)
+        : new Date(datetimeString).toLocaleDateString();
+
+    return formattedDate;
+};
+
+const format1DTo2DArray = (
+    inputArray: ParcelClients[],
+    elementsPerSubarray: number
+): ParcelClients[][] => {
+    const result = [];
+
+    for (let index = 0; index < inputArray.length; index += elementsPerSubarray) {
+        result.push(inputArray.slice(index, index + elementsPerSubarray));
```

Comment:
Also just a quick Q, how does this work on the last slice? Like when it goes over the final index does it just go to the end?

Reply:
I'm always an advocate of 'try and it see'!
![image](https://github.com/VauxhallFoodbank/vauxhall-foodbank/assets/104017547/ed855b30-52ea-43ca-bdb0-41325f6301d2)